app/javascript/controllers/flash_controller.js

```diff
@@ -0,0 +1,25 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class AdminFlashController extends Controller {
+  static targets = ['close'];
+
+  setCookie() {
+    const flashData = this.closeTarget;
+    const messageId = flashData.dataset.id;
+    const messageExpiration = flashData.dataset.expire;
+    this.setAdminFlashCookie(messageId, 'disabled', messageExpiration);
+  }
+
+  setAdminFlashCookie(name, value, date) {
+    const cookieDate = this.convertToCookieDate(date);
+    document.cookie = `admin_flash_${name}=${value}; expires=${cookieDate}; path=/`;
+  }
+
+  convertToCookieDate(date) {
+    const parts = date.split('-');
```

Comment:
I think we can simplify this date logic a lot. First we can pass in a better value from the expiresAt data value by using `to_datetime`:
```
data-admin-flash-expires-at-value="<%= admin_flash.expires.to_datetime %>"
```

then in the controller, we can do it all in a one liner that will allow us to remove the `convertToCookieDate` function:
```javascript
setCookie() {
    const expiresAt = new Date(this.expiresAtValue).toUTCString();

    document.cookie = `admin_flash_${this.idValue}=disabled; expires=${expiresAt}; path=/`;
  }